v3/src/transports/http/handler.ts

```diff
@@ -0,0 +1,145 @@
+import { IncomingMessage, RequestListener, ServerResponse } from "http";
+import { IHttpRequest } from "./transport";
+import { processHttpRequest } from "./transport";
+import { GraphQLSchema } from "graphql";
+import { GraphQLRequest } from "../../types";
+
+/**
+ * A factory function that receives an instance of `ApolloServer` and returns
+ * a `RequestHandler` that can be used with Node.js' `http.createServer`, or
+ * Express' `app.use`.
+ *
+ * @param apollo An instance of `ApolloServer`
+ */
+export const httpHandler: (
+  schema: GraphQLSchema,
```

Comment:
This binds the handler to a particular `schema`, which it seems like we may not want? In particular, what happens if the schema changes (e.g. due to a federated context push)? (We're not using this in the gateway rn, but I think ideally we would?)

Also, as I'm thinking about writing other transports… it seems like what I'd really like to receive as a transport implementer is just an `execute` function. Not GraphQL's execute, but something simpler, like:

```ts
function execute<Ctx=void>(
  query: string,
  variables: Record<string, any>,
  transportContext?: Ctx // ⬅ Execute will put this
                         // on some well-defined location
                         // within the request context, like ctx.transport
): ExecutionResponse<Ctx>

// where:
interface ExecutionResponse<Ctx> {
  transportContext: Ctx // ⬅ The same transportContext
                         // we passed in, so not strictly necessary,
                         // but I think nice for interface usability?
                         // (i.e. we can call `execute` with a 
                         // new HttpTransportContext` and not
                         // worry about it).
  responses: AsyncIterable<GraphQLResponse>
  // …and maybe some other stuff in the future, like a delta stream
}
```

This shifts the burden of dealing with schema changes (and schema *anything*, actually) to whoever is creating the transport; the `execute` function could remain stable even in the face of schema changes.

wyt?

Reply:
I agree!  And I think the first iteration of this was doing almost precisely what you're suggesting here in that it accepted `apollo`, but really only cared that `apollo.executeOperation` existed and executed an operation.

In order to land this, I changed the expectations of this function to rely on `schema` which is what was made available in the new execution work.  (The new execution work avoids the `ApolloServer` class.)

You can roughly see what the state of that first iteration was in [this code which has been removed in this same PR](https://github.com/apollographql/apollo-server/pull/3576/files#diff-5365b5da7a1792f8d0725fe342ac8a2fL273-L276).

Reply:
No, the `schema` arg isn't missing—the idea is that the transport doesn't care about the schema, it just needs a function that can run queries.

If we did things this way, the server could provide a bound `execute` method that the user can give to transports:

```ts
const server = new ApolloServer({ /* ... */ })
app.use('/graphql', httpHandler(server.execute));
```

Or we might decide it's friendlier to give the server a method that injects a transport:

```ts
app.use('/graphql', server.transport(httpHandler))
```

In a Liftoff world, we'd make a ref for the transport that injects `httpHandler` with `execute`:

```ts
// given:
export const Execute = func <typeof execute> `Execute a GraphQL query` ()

// we can do:
const HttpTransport = Execute.map(httpHandler)
const server = Apollo(() => { /* ... */ })
app.use('/graphql', server(HttpTransport))
```

(As of #3571 `Apollo` returns a `Promise<ApolloServer>` rather than a ref getter, but that'll change.)

Reply:
Also, we could call this something other than `execute` to avoid confusion with the GraphQL function of the same name that does take a schema. `run`? `query`? `operate`?